quarkus/service/src/main/java/org/apache/polaris/service/quarkus/config/QuarkusFeaturesConfiguration.java

```diff
@@ -28,6 +29,13 @@ public interface QuarkusFeaturesConfiguration extends FeaturesConfiguration {
   @Override
   Map<String, String> defaults();
 
+  @WithParentName
```

Comment:
I _think_ this line change goes too far; basically with this annotation we go from:

```
polaris.features.realm-overrides."my-realm"."feature"=true
```

to:

```
polaris.features."my-realm"."feature"=true
```

Which is definitely more concise, but I wonder if SmallRye will be smart enough to "understand" that "my-realm" is a realm ID and not a feature configuration name 🤔 

@eric-maynard did you validate that this works? If it does, I'm OK to do this but we'd need to revisit documentation + helm chart.

Reply:
I validated that the service starts up with:

```
polaris.features."my-realm"."INITIALIZE_DEFAULT_CATALOG_FILEIO_FOR_TEST"=true
```

But you're right that SmallRye may just think `my-realm` is a feature... actually, I don't see how it would be differentiating. Since it seems that don't have a test for this, let's play it safe and add the `realm-overrides` back in

Reply:
Well it _might_ be able to differentiate by the number of available path segments, e.g. the two lines below are (at least for us humans) unambiguously defined to be a default for the first one, and an override for the second one:

```properties
polaris.features.INITIALIZE_DEFAULT_CATALOG_FILEIO_FOR_TEST"=true
polaris.features."my-realm"."INITIALIZE_DEFAULT_CATALOG_FILEIO_FOR_TEST"=false
```

But I agree that it may be a bit risky to let a machine figure this subtlety out.

Reply:
From what I've read, it will first check all the explicitly-named parts of the structure, and then put the remainder in the `Map<String, ...> defaults`. From my testing, it works now. I copied a similar test to the one on @gh-yzou's PR here.